### DISCUSSION TITLE
Implicit/Explicit methods for casting numeric types

### DISCUSSION THREAD

**Discussion by LupusInferni315** [dotnet/runtime#80768] [General]
Most classes/structs that allow conversion declare an implicit or explicit operator that allows such conversion to take place, but numeric types (which can in cases be implicitly, and can be explicitly cast) don't offer implementation in the c# representation of the data types.

Unless I'm terribly mistaken, I know that numeric conversion 'occurs' in IL, but it would be useful if .NET 'exposed' the conversion methods.

an example case where this would be helpful comes from a project I'm working on where I've created a couple methods

```csharp
// 23 lines
public static bool IsCastableFrom(this Type from, Type to);

// 33 lines
public static bool TryCast<T>(this object obj, out T convertedObj);

// 961 lines
private static bool TryCastNumericType<T>(this object obj, out object convertedObj);
```

That last method checks which type is being converted from and which type is being converted to and manually performs the conversion, while for every other type, its a matter of getting a hold of the conversion method and invoking it. If the implementation of c# included exposed methods for numeric conversion, this code, and the other places I've seen 'magic' methods like this, wouldn't have to exist.

**Comment by CyrusNajmabadi** [comment]
Afaict, this seems to be an API request and not something about c#. Would you like me to move his to the runtime repo?

**Reply by LupusInferni315** [reply]
Yes please, I can never tell where my suggestions should go.
